PubSub/src/PubSubClient.php

```diff
@@ -93,10 +94,10 @@
  */
 class PubSubClient
 {
-    use ArrayTrait;
     use ClientTrait;
     use IncomingMessageTrait;
     use ResourceNameTrait;
+    use GapicClientTrait;
```

Comment:
Is it a good idea to do this here? PubSubClient is not really a GAPIC client.

Reply:
The only methods we use from `GapicClientTrait` are `buildClientOptions` and `createCredentialsWrapper`, so we could potentially split these out in a separate Trait. I suggest something like `ClientOptionsTrait`, as the `PubSubClient` IS now something that supports the same client options (even if it's not GAPIC).

Another approach we could take is to expose a public `getProjectId` function on `GapicClientTrait`, and then call it from `PubSubClient`. This would change how things work today a little bit - we'd want to create all the Gapic clients up front.

One advantage of how this is being done here is the client options (and so also, the `Credential` object) are only being created once, and then they're passed down to the gapic clients. This is a slightly more efficient design than having the Gapic clients determine their own client options/credentials each time.